doc/adr/0009-js-idle-heartbeat.md

```diff
@@ -0,0 +1,52 @@
+# 9. js-idle-heartbeat
+
+Date: 2021-06-30
+
+## Status
+
+Accepted
+
+## Context
+
+The JetStream ConsumerConfig option `idle_heartbeat` enables server-side
+heartbeats to be sent to the client. To enable the option on the consumer simply
+specify it with a value representing the number of nanoseconds that the server
+should use as notification interval.
+
+The server will only notify after the specified interval has elapsed and no new
+messages have been delivered to the consumer. Delivering a message to the
+consumer resets the interval.
+
+The idle heartbeats notifications are sent to the consumer's subscription as a
+regular NATS message. The message will have a `code` of `100` with a
+`description` of `Idle Heartbeat`. The message will contain additional headers
```

Comment:
might be worth calling out that the `description` isnt actually a promise or part of the API, the code is of course.